leaf-ui/rappid-extensions/BackboneModelsIntention.js

```diff
@@ -39,17 +38,526 @@ var EvolvingFunctionBBM = Backbone.Model.extend({
+        // Original Function
+        // for (var i = 0; i < this.functionSegList.length; i++) {
+        //     if (this.functionSegList[i] instanceof RepFuncSegment) {
+        //         var len = this.functionSegList[i].functionSegList.length;
+        //         return this.functionSegList[i].functionSegList[len - 1].funcStop;
+        //     }
+        // }
+    },
+
+    /**
+     * Deleted Functions in EvolvingFunctionBBM
+     * All of these functions were either never used or are now not needed b/c of refactor
+     * 
+     * getRepeatAbsTime: function() {},
+     * getRepeatRepNum: function() {},
+     * hasRepeat: function() {}, 
+     * setAbsoluteTime: function(time) {},
+     * setRepNum: function(num) {},
+     * getFuncSegmentIterable: function() {},
+     * getRepFuncSegmentIndex: function() {},  
+     * getLastStopValue: function() {},
+     * getMarkedVal: function(i) {},  
+     * findSegmentByStartTime: function(time) {}, 
+     */
+
+
 });
 
 var IntentionBBM = Backbone.Model.extend({
     initialize: function(options) { 
         this.nodeType = options.nodeType;  
     }, 
     defaults: { 
-        nodeName: "untitled",
+        nodeName: 'untitled',
         nodeActorID: null,                     // Assigned on release operation.
         evolvingFunction: null, 
         initialValue: '(no value)'
-    }
+    }, 
+ 
+    //will likely have to change this function 
+    changeInitialSatValue: function(initValue) {
+        var intentionEval = analysisRequest.getUserEvaluationByID(this.get('id'), '0');
+        intentionEval.evaluationValue = initValue;
+ 
+        // if there is only one function segment, and its constant, then we need to
+        // change the function segment's marked value
+ 
+        var funcSegList = this.evolvingFunction.get('functionSegList');
+        
+        if (this.evolvingFunction.get('stringDynVis') == 'C' || 
+            (this.evolvingFunction.get('stringDynVis') == 'UD' && funcSegList[0].get('type') == 'C')) { 
+                functionSegList[0].set('refEvidencePair', initValue); 
+            }
+        this.evolvingFunction.set('stringDynVis', 'NT');
+        this.evolvingFunction.set('functionSegList', []);
+    }, 
+ 
+    /**
+     * Sets the initial satisfaction value for this Intention to '(no value)'
+     */
+    removeInitialSatValue: function() {
+        this.changeInitialSatValue('(no value)');
+    },    
+
+    /**
+     * Sets newID as the new ID for this intention
+     */
+    // TODO - update this function
+    // if we are using the automatically generated 'id', then we can just call 
+    // it whenever we need?? and would we be not allowed to change it??
+    setNewID: function(newID) {
+        var oldID = this.get('id');   
+        this.set('id', newID); 
+        this.evolvingFunction.set('id', newID); 
+ 
+        // UserAssignementsList
+        var userAssignList = analysisRequest.userAssignmentsList;
```

Comment:
We don't want to access the analysisRequest global variable since we are trying to phase it out. In the future the userAssignmentsList will be stored on the graph object.